### PR TITLE
Remove master default

### PR DESCRIPTION
**Pull request by gvonkoss**
This removes the hard default to master, which is a limitation for some of the components that don't necessarily require the use of that brand (such as o-header-services, o-layout, etc).

However, this means that a component has to define one of the available brands (master/internal/whitelabel) in order to be able to use o-brand. 

I've also added an error helper to be able to test the errors that get output, and i've broken up the file structure because for the longest time I thought that everything in the mixin folder was a mixin and it made my logic not right 😬

### CODE REVIEW

File: test/scss/index.test.scss [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 @import '../../main';
 
+$_test-environment: true;
```
[the remark this answers is not in the document]
Comment (gvonkoss) [reply test/scss/index.test.scss]: in `_oBrandError` for help with testing

**Comment by gvonkoss** [comment]
It wouldn't. `o-brand` as is currently requires `master` to be defined or it will throw an error, whereas this change requires any of the supported brands to be defined, or it throws an error. It won't break the branded components, but it will allow us to hide away warnings for brands that are being defined but not used 🤞

**Comment by notlee** [comment]
👍

Review incoming, this is noting down our in person discussion:

There's a risk of a breaking release as it will error if a component does not have the internal or whitelabel brand defined, we should release a major or, as o-brand is for internal use only, do a thorough review of our component branding.

- Should error if an unrecognised brand is requested.
- Error if the component does not support the brand.
- Set one brand for a component (i.e. just internal).
    - Should not error if masterbrand is not defined.
- Defaults to master on a global level.

File: README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ Tools to tailor Origami components for distinct usecases.
 
 ### Brand
 
-A brand represents an enviroment which requires components to offer a distinct appearence or unique functionality.
+A brand represents an environment which requires components to offer a distinct appearance or unique functionality.
```
Comment (notlee) [inline README.md]: 😬 nice spot

**Comment by gvonkoss** [comment]
Closing this in favour of https://github.com/Financial-Times/o-brand/pull/28/ now that the actual change has been merged.

**Pull request closed without merging**